From 44fd8bf56674a59a512d6abc061288723912f3dd Mon Sep 17 00:00:00 2001
From: Andy Fleming <afleming@freescale.com>
Date: Wed, 7 Oct 2009 03:32:02 -0500
Subject: [PATCH 043/252] dpa: Don't initialize disabled devices

Device nodes might have a "status" property, declaring the device as
"disabled", and if so, we should not initialize the device.

Signed-off-by: Andy Fleming <afleming@freescale.com>
[Cleanly applied the FSL SDK 2.0.3 patch:
"kernel-2.6.30-dpa-Don-t-initialize-disabled-devices.patch"]
Integrated-by: Yuri Nedel <Yuri.Nedel@windriver.com>
---
 drivers/net/dpa/dpa.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/net/dpa/dpa.c b/drivers/net/dpa/dpa.c
index c028f69..10376a2 100644
--- a/drivers/net/dpa/dpa.c
+++ b/drivers/net/dpa/dpa.c
@@ -2064,6 +2064,9 @@ dpa_init_probe(struct of_device *_of_dev)
 
 	dpa_node = _of_dev->node;
 
+	if (!of_device_is_available(dpa_node))
+		return -ENODEV;
+
 	/* FM */
 
 	mac_dev = dpa_mac_probe(_of_dev);
@@ -2212,6 +2215,7 @@ dpa_probe(struct of_device *_of_dev)
 	const uint32_t		*rx_fqids;
 	int			num_tx_fqids, num_tx_fqs;
 	int			num_rx_fqids, num_rx_fqs;
+	const char *dpa_status;
 
 	dev = &_of_dev->dev;
 
@@ -2219,6 +2223,9 @@ dpa_probe(struct of_device *_of_dev)
 
 	dpa_node = _of_dev->node;
 
+	if (!of_device_is_available(dpa_node))
+		return -ENODEV;
+
 	/*
 	 * Allocate this early, so we can store relevant information in
 	 * the private area
-- 
1.6.5.2

